### PR TITLE
✨ [Chrome Attribution Report API] Add attribution reporting to amp-pixel

### PR DESCRIPTION
**Pull request by emensc52**
This PR will enable amp-pixel tag support for Chrome Attribution Reporting API by carrying over the attributionsrc attribute to the generate img request.

/cc @calebcordry

### CODE REVIEW

File: src/builtins/amp-pixel/amp-pixel.js [hunk 1]
```diff
@@ -18,6 +18,10 @@ export class AmpPixel extends BaseElement {
 
     /** @private {?Promise<!Image>} */
     this.triggerPromise_ = null;
+
+    /** @private @const {boolean} */
+    this.isAttributionReportingSupported_ =
```
Comment (calebcordry) [inline src/builtins/amp-pixel/amp-pixel.js:23]: I just made #38620 where I try to pull this util out to be shared. Once I get that in we can also use here.

File: src/pixel.js [hunk 2]
```diff
@@ -10,27 +10,35 @@ const TAG = 'pixel';
...
  * @return {!Element}
  */
-function createNoReferrerPixel(win, src) {
+function createNoReferrerPixel(win, src,
+                                attributionReportingEligible,
+                                attributionSrc) {
   if (isReferrerPolicySupported()) {
-    return createImagePixel(win, src, true);
+    return createImagePixel(win, src, true, attributionReportingEligible, attributionSrc);
   } else {
```
Comment (calebcordry) [inline src/pixel.js:42]: note I think we only get here on ie11 which we dont actually support anymore

File: src/pixel.js [hunk 3]
```diff
@@ -54,15 +62,22 @@ function createNoReferrerPixel(win, src) {
  * @param {!Window} win
  * @param {string} src
  * @param {boolean=} noReferrer
+ * @param {?boolean=} attributionReportingEligible
+ * @param {string=} attributionsrc
  * @return {!Image}
  */
-function createImagePixel(win, src, noReferrer = false) {
+function createImagePixel(win, src, noReferrer = false,
```
Comment (calebcordry) [inline src/pixel.js:69]: if we do the check in this func I think we can get rid of  `attributionReportingEligible` param

**Review by calebcordry — COMMENTED** [review]
Thanks for making this PR! We will also need to add a validator rule to allow this attr see https://github.com/ampproject/amphtml/blob/fc447110e86db57e4391bf5c50de8dd2bf75e18f/docs/component-validator-rules.md and https://github.com/ampproject/amphtml/blob/main/validator/validator-main.protoascii#L4045

**Comment by calebcordry** [comment]
Partial https://github.com/ampproject/amphtml/issues/35347

File: src/pixel.js (hunk 2 above)
Comment (emensc52) [reply src/pixel.js:42]: Do you mean the else condition where referrer policy is not supported? Is this confirmed? I just remembered we have to add allow='attributing-reporting' for img tags generated on iframes. But this is the only place we generate iframes for pixels, so if this is only ie11, we don't have to add it and can even remove adding attributionSrc in this condition as we only care about Chrome browers.

File: src/builtins/amp-pixel/amp-pixel.js (hunk 1 above)
Comment (emensc52) [reply src/builtins/amp-pixel/amp-pixel.js:23]: Thanks! Let me know when that change is made and I'll update this.

File: src/pixel.js (hunk 2 above)
Comment (calebcordry) [reply src/pixel.js:42]: https://caniuse.com/mdn-html_elements_img_referrerpolicy seems if all we care about is chrome we are safe.

**Comment by emensc52** [comment]
> Thanks for making this PR! We will also need to add a validator rule to allow this attr see https://github.com/ampproject/amphtml/blob/fc447110e86db57e4391bf5c50de8dd2bf75e18f/docs/component-validator-rules.md and https://github.com/ampproject/amphtml/blob/main/validator/validator-main.protoascii#L4045

@calebcordry Done! Please review the new validator files.

File: src/pixel.js (hunk 2 above)
Comment (emensc52) [reply src/pixel.js:39]: Nice, I removed the attributionsrc parameter in the non-referrer case.

File: src/pixel.js (hunk 3 above)
Comment (emensc52) [reply src/pixel.js:66]: Done.

File: src/pixel.js [hunk 4]
```diff
@@ -43,7 +48,7 @@ function createNoReferrerPixel(win, src) {
       }
     );
     iframe.onload = () => {
-      createImagePixel(iframe.contentWindow, src);
+      createImagePixel(iframe.contentWindow, src, false);
```
Comment (calebcordry) [inline src/pixel.js:51]: do we need this change?

File: validator/testdata/amp4ads_feature_tests/amp_pixel_ar.html [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+<html ⚡4ads>
+<head>
+  <meta charset="utf-8">
+  <meta name="viewport" content="width=device-width,minimum-scale=1">
+  <style amp4ads-boilerplate>body{visibility:hidden}</style>
+  <script async src="https://cdn.ampproject.org/amp4ads-v0.js"></script>
+</head>
+<body>
+  Hello, world.
+  <amp-pixel src="https://pixel.me?yes=please" attributionsrc=""></amp-pixel>
```
Comment (calebcordry) [inline validator/testdata/amp4ads_feature_tests/amp_pixel_ar.html:15]: can we put another `<amp-pixel>` with https url

File: src/pixel.js (hunk 4 above)
Comment (emensc52) [reply src/pixel.js]: Nope, lingering change. Removed.

File: validator/testdata/amp4ads_feature_tests/amp_pixel_ar.html (hunk 5 above)
Comment (emensc52) [reply validator/testdata/amp4ads_feature_tests/amp_pixel_ar.html:15]: Yep, added.

File: src/builtins/amp-pixel/amp-pixel.js (hunk 1 above)
Comment (calebcordry) [reply src/builtins/amp-pixel/amp-pixel.js]: This was merged today so you should be able to use it. Note you may have to allowlist the dep check (see my PR)

**Comment by calebcordry** [comment]
Adding @amaltas for validator review. 

Also looks like a few lint errors, you can check with `amp lint --local_changes`.

File: src/builtins/amp-pixel/amp-pixel.js (hunk 1 above)
Comment (calebcordry) [reply src/builtins/amp-pixel/amp-pixel.js]: If they pass you should be good.

**Review by calebcordry — APPROVED** [review]

**Comment by calebcordry** [comment]
> Also looks like a few lint errors, you can check with amp lint --local_changes

Still a few of these https://app.circleci.com/pipelines/github/ampproject/amphtml/26838/workflows/3015fee6-83ca-4946-80bd-350fde77db9a/jobs/561045

**Review by amaltas — APPROVED** [review]
Approve validator changes

**Comment by calebcordry** [comment]
@erwinmombay can you take a look for `src/`, we had to move some files there for this PR as amp pixel is a built-in

**Review by erwinmombay — APPROVED** [review]

**Comment by erwinmombay** [comment]
@calebcordry LGTM

**Pull request merged**